pkg/controller/certificates/sync.go

```diff
@@ -442,7 +442,12 @@ func (c *Controller) updateSecret(ctx context.Context, crt *v1alpha1.Certificate
 	secret.Annotations[v1alpha1.IPSANAnnotationKey] = strings.Join(pki.IPAddressesToString(x509Cert.IPAddresses), ",")
 
 	// Always set the certificate name label on the target secret
-	secret.Labels[v1alpha1.CertificateNameKey] = crt.Name
+	// Get the shortened label version of the certificate name
+	labelValue, err := util.NameToLabel(crt.Name)
+	if err != nil {
+		return nil, err
+	}
+	secret.Labels[v1alpha1.CertificateNameKey] = labelValue
```

Comment:
I wonder how useful it is for the Secret to have the hashed value as a label. Would we be better just moving this label to be an annotation?

We probably also want to continue setting this label if it *is* possible to do, as some users may have developed tools and scripts that rely on it. I think maybe we can allow users to toggle the old "set label" behaviour using a feature flag for `DeprecatedSetTruncatedSecretLabel`, which will default to `true` in this release and in a future release can be defaulted to `false` before being removed.

Reply:
I think what I will do is keep the label (but truncated **without hash**) for compatibility, and use `OwnerReferences` for clean up - an issue with this is that we cannot use a selector or similar means (https://github.com/kubernetes/kubernetes/issues/54498), however this is probably preferred over exposing a label which is only helpful internally.

Reply:
Thinking about it more, we may not want to set `OwnerReferences` on the Secret in order to prevent garbage collection on it (there is flag which disables this so I assume it is generally not desirable). Therefore I imagine we still require some sort of label on the Secret in order to identify its owning Certificate, perhaps the one for the truncated name (for compatibility) and one for a hashed version of the name (required to be a label to order to use Selector)?

Reply:
We can use annotations here instead of labels, which I think will resolve the majority of these issues. The new Certificate controller implementation I'm putting together already does this instead of Secrets.

> we may not want to set OwnerReferences on the Secret in order to prevent garbage collection on it

Correct 😄